package/MDAnalysis/coordinates/DCD.py

```diff
@@ -1,90 +1,37 @@
-# and contributors (see AUTHORS for the full list)
+# based on MDAnalysis.coordinates.DCD from
 #
+# MDAnalysis --- http://mdanalysis.googlecode.com
+# Copyright (c) 2006-2011 Naveen Michaud-Agrawal,
+#               Elizabeth J. Denning, Oliver Beckstein,
+#               and contributors (see website for details)
 # Released under the GNU Public Licence, v2 or any higher version
 #
 # Please cite your use of MDAnalysis in published work:
 #
-# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
-# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
-# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#     N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and
+#     O. Beckstein. MDAnalysis: A Toolkit for the Analysis of
+#     Molecular Dynamics Simulations. J. Comput. Chem. 32 (2011), 2319--2327,
+#     doi:10.1002/jcc.21787
 #
 
-
-"""DCD trajectory I/O  --- :mod:`MDAnalysis.coordinates.DCD`
-============================================================
-
-Classes to read and write DCD binary trajectories, the format used by
-CHARMM, NAMD, and also LAMMPS. Trajectories can be read regardless of
-system-endianness as this is auto-detected.
-
-Generally, DCD trajectories produced by any code can be read (with the
-:class:`DCDReader`) although there can be issues with the unitcell
-(simulation box) representation (see
-:attr:`Timestep.dimensions`). DCDs can also be written but the
-:class:`DCDWriter` follows recent NAMD/VMD convention for the unitcell
-but still writes AKMA time. Reading and writing these trajectories
-within MDAnalysis will work seamlessly but if you process those
-trajectories with other tools you might need to watch out that time
-and unitcell dimensions are correctly interpreted.
-
-.. Note::
-
-   The DCD file format is not well defined. In particular, NAMD and
-   CHARMM use it differently. Currently, MDAnalysis tries to guess the
-   correct **format for the unitcell representation** but it can be
-   wrong. **Check the unitcell dimensions**, especially for triclinic
-   unitcells (see `Issue 187`_ and :attr:`Timestep.dimensions`). A
-   second potential issue are the units of time which are AKMA for the
-   :class:`DCDReader` (following CHARMM) but ps for NAMD. As a
-   workaround one can employ the configurable
-   :class:`MDAnalysis.coordinates.LAMMPS.DCDReader` for NAMD
-   trajectories.
-
-.. SeeAlso:: The :mod:`MDAnalysis.coordinates.LAMMPS` module provides
-             a more flexible DCD reader/writer.
-
-The classes in this module are the reference implementations for the
-Trajectory API.
-
-.. _Issue 187:
-   https://github.com/MDAnalysis/mdanalysis/issues/187
-
-
-Classes
--------
-
-.. autoclass:: Timestep
-   :inherited-members:
-.. autoclass:: DCDReader
-   :inherited-members:
-.. autoclass:: DCDWriter
-   :inherited-members:
-
-"""
-from __future__ import absolute_import, division, print_function, unicode_literals
+from __future__ import absolute_import
```

Comment:
Yeah, many / most of the docstrings got obliterated in the frenzy of trying to 'get it working' as well. I think Alexander had removed some content (including docstrings, Writer object, etc.) he didn't need, so the workflow was something like taking his versions of the dcd python / c modules, then restoring some critical / current MDA code to register the reader and deal with the API changes from the last few months, and then start pasting the DCD Writer and Timeseries objects back in. I suspect there is a smart way to progressively merge in changes from different repos / branches at certain line numbers only, but the extra 'git fu,' at this stage, would probably just detract from my attempts to engineer a solution.

I suspect I'll have to try to get it working first and then there will be some pretty annoying damage control on assigning credit and making sure I don't get credited for just pasting stuff that already existed, etc. Conversely, the disregard for making gradual changes in the workflow could end up making the overall task take longer in the end!